### PR TITLE
[BACKLOG-5526] Make the ServerStatusProvider pluggable.

### PR DESCRIPTION
**Pull request by tkafalas**

### CODE REVIEW

File: api/ivy.xml [hunk 1]
```diff
@@ -22,6 +22,10 @@
 
 		<dependency org="org.springframework.security"   name="spring-security-core"      rev="${dependency.spring.security.revision}" transitive="false"/>
 		<dependency org="org.springframework"            name="spring-beans"              rev="${dependency.spring.framework.revision}"         transitive="false"/>
+		
+		<dependency org="org.slf4j" name="slf4j-api" rev="1.7.3" transitive="false"/>
+    	<dependency org="org.slf4j" name="slf4j-log4j12" rev="1.7.3" transitive="false"/>
```
Comment (pentaho-nbaker) [inline api/ivy.xml]: We should only need the slf4j-api here, not the log4j stuff

File: api/src/org/pentaho/platform/api/engine/IServerStatusProvider.java [hunk 2]
```diff
@@ -37,4 +43,33 @@
...
+      if ( System.getProperty( PROVIDER_CLASS ) != null ) {
+
+        try {
+          instance = (IServerStatusProvider) Class.forName( System.getProperty( PROVIDER_CLASS ) ).newInstance();
+        } catch ( ClassNotFoundException e ) {
+          logger.error( "ServerStatusProvider class not found", e );
+        } catch ( InstantiationException | IllegalAccessException e ) {
+          logger.error( "ServerStatusProvider class could not be instantiated", e );
+        }
+        return instance;
```
Comment (pentaho-nbaker) [inline api/src/org/pentaho/platform/api/engine/IServerStatusProvider.java]: If any error occurs using their custom class we should instantiate and return the default

File: api/ivy.xml (hunk 1 above)
Comment (tkafalas) [inline api/ivy.xml]: Without the log4j stuff the ServerStatusProvider was erroring out on the logger.error

Comment (pentaho-nbaker) [inline api/ivy.xml]: Ah you mean in a test? You shouldn't need a binding SLF4J just goes silent if no binding is present:
http://www.slf4j.org/manual.html

File: api/src/org/pentaho/platform/api/engine/IServerStatusProvider.java (hunk 2 above)
Comment (tkafalas) [inline api/src/org/pentaho/platform/api/engine/IServerStatusProvider.java]: Done.

**Comment by pentaho-nbaker** [comment]
Cool thanks Tim

**Pull request merged**